gigascience/gigadb-website
# Fix admin file add, edit and delete attributes form

# Pull request for issue: #1507

This is a pull request for the following functionalities:

* Fix issues with admin file form that allows to add attributes

## How to test?

* Navigate to http://gigadb.gigasciencejournal.com:9170/adminFile/update/id/446
* Submit the form without doing anything else and verify that it works
* Click on the "show new attr..." button and click on submit the form. Verify client side validation
* Fill in new attribute fields and click on "add attribute". Verify that attribute has been added
* Verify that attribute edit works
* Verify that deleting an attribute works

## How have functionalities been implemented?

The hidden fields for new attribute had a required attribute that enabled client side validation even when hidden, making it impossible to submit the form if the dropdown is collapsed

![image](https://github.com/gigascience/gigadb-website/assets/143437854/56f469f0-5c5b-4ce3-99a0-50b3a6905bfd)

Programmatically toggling the `required` solves the issue

![image](https://github.com/gigascience/gigadb-website/assets/143437854/fc7eebc0-ad6e-411c-bd2d-c724e8813948)


## Any issues with implementation?

None

## Any changes to automated tests?

Some acceptance tests required this fix to work. After the fix they pass

## Any changes to documentation?

None

## Any technical debt repayment?

None

## Any improvements to CI/CD pipeline?

None
